File: moveZeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int countZeroes = 0;
        for(int i = 0; i < nums.length; i++){
            if(nums[i] == 0){
                countZeroes++;
            }else if(countZeroes > 0){
                int temp = nums[i];
                nums[i] = 0;
                nums[i - countZeroes] = temp;
            }
        }
    }
}

File: moveZeroes.js
var moveZeroes = function(nums){
    let countZeroes = 0;
    let temp = 0;
    for(let i = 0; i < nums.length; i++){
        nums[i] == 0 ? countZeroes++
            : (temp = nums[i],
                nums[i] = 0,
                nums[i - countZeroes] = temp);
    }
};